Formula/lean.rb

```diff
@@ -0,0 +1,35 @@
+class Lean < Formula
+  desc "Theorem prover"
+  homepage "https://leanprover.github.io/"
+  url "https://github.com/leanprover/lean/archive/v3.2.0.tar.gz"
+  sha256 "2668e075a9bdf4270d75fd60181078b33c078c7288b5bb8ad795819d8d5272cf"
+  head "https://github.com/leanprover/lean.git"
+
+  depends_on "cmake" => :build
+  depends_on "gmp"
+  depends_on "jemalloc" => :recommended
```

Comment:
We try to have formulas with the fewer options possible, so many people can benefit from our prebuilt binaries (bottles). Let's make `jemalloc` mandatory.